sdk/maps/azure-maps-geolocation/src/main/java/com/azure/maps/geolocation/GeolocationAsyncClient.java

```diff
@@ -60,7 +64,25 @@ public Mono<Response<IpAddressToLocationResult>> getLocationWithResponse(JsonFor
      * @return this object is returned from a successful call to IP Address to country/region API.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<IpAddressToLocationResult> getLocation(JsonFormat format, String ipAddress) {
-        return this.serviceClient.getLocationAsync(format, ipAddress);
+    public Mono<Response<IpAddressToLocationResult>> getLocationWithResponse(String ipAddress) {
+        return this.getLocationWithResponse(ipAddress, null);
+    }
+
+    /**
+     * **Applies to**: S0 and S1 pricing tiers.
+     *
+     * <p>This service will return the ISO country code for the provided IP address. Developers can use this information
+     * to block or alter certain content based on geographical locations where the application is being viewed from.
+     *
+     * @param format Desired format of the response. Only `json` format is supported.
+     * @param ipAddress The IP address. Both IPv4 and IPv6 are allowed.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return this object is returned from a successful call to IP Address to country/region API.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<IpAddressToLocationResult>> getLocationWithResponse(String ipAddress, Context context) {
```

Comment:
The public method of an AsyncClient cannot have the Context parameter. Please modify this so it's only accessible from within the package. Just remove "public".